### PR TITLE
Proposal to utilize Nox for runner

### PR DESCRIPTION
**Pull request by abefleming**
Proposal is a strong word, demo is more like it.

To get started, jump into poetry shell and update dependencies. 

- `nox -l` to list different sessions that are defined in the `noxfile.py`
- `nox` to run all sessions
- `nox -e <session name>` to run a specific  session
- `nox --report status.json` to get an output file

Nox is a spiritual successor to tox, which is a testing automation tool, and lets you create a testing runner that is done with Python. The designated config file is `noxfile.py` and works similarly to tox. You create sessions, and in each sessions you describe a virtual environment that will be executed. This includes the installation of dependencies as well as any arbitrary command that you can run. 

Totally wanted to try utilizing something like this and the Makefile had a really easy to translate layout, so I figured it was a good opportunity to try. 

# PROS:
- Gives benefit of orchestrating all your testing and linting tasks
- Can do pretty much anything you can spit into a subprocess Ex: https://nox.thea.codes/en/stable/cookbook.html#the-auto-release
- Written in python with no additional config required
- Cross platform OOTB 
- No issues respecting/picking up config files
- [Pip uses it](https://github.com/pypa/pip) (Other big names use it, but Pip was the main seller for me)

# CONS:
- Not as flexible as Make, where CLI magic is the main language
- Virtualenv management can be a bit finicky when it comes to dependencies, but poetry helps a lot here
- Commands in the tasks are written as lists of strings like in `subprocess.run()` which is a real PITA, but config files help here a lot as well
- You don't like the name Nox for some reason

TL;DR; Thought it was neat, check it out sometime. Wanna know what the homie think.

### CODE REVIEW

**Comment by ajmassi** [comment]
Closed following conversation. Definitely an intriguing tool and proposal, but a heavy integration for this project's simple release needs.
